baybe/utils/chemistry.py

```diff
@@ -72,158 +72,117 @@ def name_to_smiles(name: str) -> str:
     """
-    mols = smiles_to_molecules(smiles_list)
+    kwargs_fingerprint = kwargs_fingerprint or {}
+    kwargs_conformer = kwargs_conformer or {}
+
+    if encoding is SubstanceEncoding.MORGAN_FP:
+        warnings.warn(
+            f"Substance encoding '{encoding.name}' is deprecated and will be disabled "
+            f"in a future version. Use '{SubstanceEncoding.ECFP.name}' "
+            f"with 'fp_size' 1204 and 'radius' 4 instead.",
+            DeprecationWarning,
+        )
+        encoding = SubstanceEncoding.ECFP
+        kwargs_fingerprint.update({"fp_size": 1024, "radius": 4})
+
+    elif encoding is SubstanceEncoding.RDKIT:
+        warnings.warn(
+            f"Substance encoding '{encoding.name}' is deprecated and will be disabled "
+            f"in a future version. Use '{SubstanceEncoding.RDKIT2DDESCRIPTORS.name}' "
+            f"instead.",
+            DeprecationWarning,
+        )
+        encoding = SubstanceEncoding.RDKIT2DDESCRIPTORS
 
-    res = []
-    for mol in mols:
-        RDLogger.logger().setLevel(RDLogger.CRITICAL)
+    fingerprint_cls = get_fingerprint_class(encoding)
+    fingerprint_encoder = fingerprint_cls(**kwargs_fingerprint)
 
-        fingerp = GetMorganFingerprintAsBitVect(mol, radius, nBits=n_bits).ToBitString()
-        fingerp = map(int, fingerp)
-        fpvec = np.array(list(fingerp))
-        res.append(
-            {prefix + "FP_" + f"{k + 1}": dtype(bit) for k, bit in enumerate(fpvec)}
+    if fingerprint_encoder.requires_conformers:
+        mol_list = ConformerGenerator(**kwargs_conformer).transform(
+            MolFromSmilesTransformer().transform(smiles)
         )
-
-    df = pd.DataFrame(res)
+    else:
+        mol_list = smiles
+
+    features = np.concatenate(
+        [
+            _molecule_to_fingerprint_features(
+                fingerprint_encoder=fingerprint_encoder, molecule=mol
+            )
+            for mol in mol_list
+        ]
+    )
+    name = f"skfp{fingerprint_encoder.__class__.__name__.removesuffix('Fingerprint')}_"
+    col_names = [prefix + name + f for f in fingerprint_encoder.get_feature_names_out()]
+    df = pd.DataFrame(features, columns=col_names, dtype=DTypeFloatNumpy)
 
     return df
 
 
-def is_valid_smiles(smiles: str) -> bool:
-    """Test if a SMILES string is valid according to RDKit.
+def get_fingerprint_class(encoding: SubstanceEncoding) -> BaseFingerprintTransformer:
+    """Retrieve the fingerprint class corresponding to a given encoding.
 
     Args:
-        smiles: SMILES string to be tested.
+        encoding: A substance encoding.
+
+    Raises:
+        ValueError: If no fingerprint class for the specified encoding.
 
     Returns:
-        ``True`` if the provided SMILES is valid, ``False`` else.
+        The fingerprint class.
     """
+    # Exception case
+    if encoding is SubstanceEncoding.RDKITFINGERPRINT:
+        return skfp_fingerprints.RDKitFingerprint
```

Comment:
@Scienfitz: this is what I meant

Reply:
sorry I need more context

Reply:
the fact that we require this exception rule and break the general pattern, as I mentioned in the call. As I said, if you say this is the most reasonable way, then this is all fine for me. I just wanted to raise awareness since this now really catches attention as the exception has become explicit with the latest changes

Reply:
so now that the deprecation is accepted I would consider this resolved unless an alternative for the exception is proposed?